v3/generate.go

```diff
@@ -358,6 +382,64 @@ func generateLMSConnectAPIYml(i map[interface{}]interface{}, version string) ([]
 	return yaml.Marshal(m)
 }
 
+// generateAttendanceAPIYml generates the Attendance API from the base yml for a specific version. It does
+// this by removing things from the yml, for example the non-Attendance endpoints.
+func generateAttendanceAPIYml(i map[interface{}]interface{}, version string) ([]byte, error) {
```

Comment:
thought: this order of doing things is very odd to me. why do we write a big file and then pick it apart into smaller pieces instead of writing smaller pieces that can then be composed/merged into fuller representations of the platform across multiple APIs? this would also give us the ability to, for example, just use the attendance-api `swagger.yml` file. it would also potentially allow us to do a more natural representation of differences across v3.0 and v3.1 and just merge different representations where different rather than using so many conditionals.

anyway, clearly outside the scope of this PR, but food for thought!

Reply:
yeah I agree, the way this was originally written is pretty hard to read 💀 can timebox a bigger refactor and/or make a ticket for it in the backlog

Reply:
one thing to note though is that this repo is public so could also be why we don't pull from `swagger.yml` files from other repos

Reply:
makes sense to me!